src/cmd/root.go

```diff
@@ -0,0 +1,104 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/tkhq/go-sdk/pkg/store"
+	"github.com/tkhq/go-sdk/pkg/store/local"
+)
+
+var (
+	rootKeysDirectory string
+
+	keyStore store.Store
+
+	// KeyName is the name of the key with which we are operating.
+	KeyName string
+
+	// Organization is the organization ID to interact with.
+	Organization string
+)
+
+func init() {
+	rootCmd.PersistentFlags().StringVarP(&rootKeysDirectory, "keys-folder", "d", local.DefaultKeysDir(), "directory in which to locate keys")
+	rootCmd.PersistentFlags().StringVarP(&KeyName, "key-name", "k", "default", "name of API key with which to interact with the Turnkey API service")
+
+	rootCmd.PersistentFlags().StringVar(&Organization, "organization", "", "organization ID to be used")
+}
+
+func basicSetup(cmd *cobra.Command) {
+	// No non-JSON-formatted output should flow over stdin; thus change
+	// output for usage messages to stderr.
+	cmd.SetOut(os.Stderr)
+
+	if keyStore == nil {
+		localKeyStore := local.New()
+
+		if err := localKeyStore.SetKeysDirectory(rootKeysDirectory); err != nil {
+			OutputError(errors.Wrap(err, "failed to obtain key storage location"))
+		}
+
+		keyStore = localKeyStore
+	}
+}
+
+// Execute runs the cobra command for the Turnkey CLI.
+func Execute() error {
+	return rootCmd.Execute()
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "turnkey interacts with the Turnkey API",
+	Short: "turnkey is the Turnkey CLI",
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		basicSetup(cmd)
+	},
+}
+
+// RequestTimestamp returns a timestamp formatted for inclusion in a request.
+func RequestTimestamp() *string {
+	ts := fmt.Sprintf("%d", time.Now().UnixMilli())
+
+	return &ts
+}
+
+// ParameterToReader converts a commandline parameter to an io.Reader based on its syntax.
+// Values of "-" return stdin.
+// Values beginning with "@" return the file with name following the "@".
+// Other values are interpreted literally.
```

Comment:
This is suuuper sweet.

Reply:
I stole it from Talos.  I forget who came up with this scheme for `talosctl`, but it's _awfully_ convenient.